### PR TITLE
Enable Parquet reader to use SeekableInputStream

### PR DESCRIPTION
**Pull request by leetcode-1533**
## Description
Enable Parquet reader to "seek" in the input stream. Part of https://github.com/trinodb/trino/issues/9471
When implementing Bloomfilter reader for parquet, as specified in https://github.com/apache/parquet-mr/commit/806037c080dc477798d157cd4a54a81240a85d37#diff-227f2b038d111d090ba898611889833804545f5c1e5c2d88a43a98a05b2970eb. 

The bloom_filter_offset in thrift specified the "offset" of the bloomfilter header, but it does not specify the "length" of the bloomfilter header. (https://github.com/apache/parquet-format/blob/master/BloomFilter.md) 

However, the Thrift generated java code can read in only necessary amount of data, when the stream starting at the "offset", in Parquet MR, this was done at: https://github.com/apache/parquet-mr/blob/master/parquet-hadoop/src/main/java/org/apache/parquet/hadoop/ParquetFileReader.java#L1303 which used SeekableInputStream.

I am also open to refactoring the suggestion of removing org.apache.iceberg.io.SeekableInputStream from the TrinoInput interface. 

I also removed HdfsParquetDataSource.java, which is currently only used by unit test.

Without this change, I need to do multiple reads until successfully serialized the object, like Impala code using binary search: https://github.com/apache/impala/blob/522ee1fcc09d47074c75440ba4fc3d258e1c95b3/be/src/exec/parquet/hdfs-parquet-scanner.h#L894. 

## Non-technical explanation
Enable Parquet reader to "seek" in the input stream.

## Release notes

(x) This is not user-visible or docs only and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

```markdown
# Section
* Fix some things. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by sopel39** [comment]
> The bloom_filter_offset in thrift specified the "offset" of the bloomfilter header, but it does not specify the "length" of the 

Since you don't know length, how would you control amount of data fetched from input stream? Is it arbitrary?

Does bloom filter have predictable size that we can use?
Can you read in chunks (starting from given `offset`) until you see end of filter?

File: plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/TrinoParquetDataSource.java [hunk 1]
```diff
@@ -64,4 +66,31 @@ protected void readInternal(long position, byte[] buffer, int bufferOffset, int
...
+    {
+        private final org.apache.iceberg.io.SeekableInputStream delegate;
+
+        private ParquetInputStreamAdapter(org.apache.iceberg.io.SeekableInputStream delegate)
+        {
+            super(delegate);
+            this.delegate = delegate;
+        }
+
+        public long getPos() throws IOException
```
Comment (electrum) [inline plugin/trino-hive/src/main/java/io/trino/plugin/hive/parquet/TrinoParquetDataSource.java:86]: Put throws clause on next line

**Review by electrum — COMMENTED** [review]
Looks good to me. Can you put the removal of HdfsParquetDataSource in a separate commit?

**Comment by leetcode-1533** [comment]
@raunaqmorarka   is there some performance penalty when doing streaming read APIs?

**Comment by raunaqmorarka** [comment]
> @raunaqmorarka is there some performance penalty when doing streaming read APIs?

There is a description of the problems encountered with streaming reads in https://trino.io/blog/2019/05/06/faster-s3-reads.html

**Pull request closed without merging**